### PR TITLE
Open telemetry support

### PR DESCRIPTION
**Pull request by DGraveling**

### CODE REVIEW

**Comment by Kielek** [comment]
@martinjt, could you please check as you have started this topic on slack?

File: src/ServiceConnect.Telemetry/ServiceConnectActivitySource.cs [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+
+    public static readonly string PublishActivitySourceName = ActivitySourceName + ".Publish";
+    public static readonly string ConsumeActivitySourceName = ActivitySourceName + ".Consume";
+    public static readonly string SendActivitySourceName = ActivitySourceName + ".Send";
+
+    private static readonly ActivitySource _publishActivitySource = new(PublishActivitySourceName, Version?.ToString() ?? "0.0.0");
+    private static readonly ActivitySource _consumeActivitySource = new(ConsumeActivitySourceName, Version?.ToString() ?? "0.0.0");
+    private static readonly ActivitySource _sendActivitySource = new(SendActivitySourceName, Version?.ToString() ?? "0.0.0");
+
+    public static Activity? Publish(PublishEventArgs eventArgs, ActivityContext linkedContext = default)
```
[the remark this answers is not in the document]
Comment (DGraveling) [reply src/ServiceConnect.Telemetry/ServiceConnectActivitySource.cs:21]: Renamed as suggested

File: src/ServiceConnect.Telemetry/ServiceConnectActivitySource.cs [hunk 2]
```diff
@@ -0,0 +1,240 @@
+﻿using System.Diagnostics;
+using System.Text;
+
+namespace ServiceConnect.Telemetry;
+
+public static class ServiceConnectActivitySource
```
[the remark this answers is not in the document]
Comment (dgraveling1) [reply src/ServiceConnect.Telemetry/ServiceConnectActivitySource.cs:6]: Hi @Kielek , any thoughts on this?

[the remark this answers is not in the document]
Comment (Kielek) [reply src/ServiceConnect.Telemetry/ServiceConnectActivitySource.cs:6]: Fully remove options class?

Following options can be handled by separate ActivitySources (you have it).

```
    public bool EnablePublishTelemetry { get; set; } = true;
    public bool EnableConsumeTelemetry { get; set; } = true;
    public bool EnableSendTelemetry { get; set; } = true;
```


You will not have this options, is it really required?
```
    public Action<Activity, Message> EnrichWithMessage { get; set; }
    public Action<Activity, byte[]> EnrichWithMessageBytes { get; set; }
```

**Pull request merged**